pyfstat/core.py

```diff
@@ -1059,128 +1084,118 @@ def plot_twoF_cumulative(
-            Colour
+        signal_parameters: dict
+            Dictionary with parameters at which to predict Fstat.
+            At least (h0, cosi, psi, Alpha, Delta, F0) keys must be
+            present (any other key will simply be ignored).
+            Alternatively, these can be read from a ".loudest" file,
+            see _calculate_predict_fstat_cumulative() for details.
+        predict_fstat_segments : int
+            Number of points to use for PredictFStat.
+        custom_axis_kwargs : dict
+            Optional axis formatting options.
+        plot_label : str
+            Legend label for the cumulative 2F values computed from data.
         savefig : bool
-            If true, save the figure in outdir
-        title, plt_label: str
-            Figure title and label
+            If true, save the figure in outdir and return taus, twoFs.
+            If false, return axes object.
+        calculate_twoF_cumulative_kwargs : dict
+            Will be passed on to calculate_twoF_cumulative().
 
         Returns
         -------
-        tauS, tauF : ndarray shape (N,)
-            If savefig, the times and twoF (cumulative) values
+        taus : ndarray of shape (predict_fstat_segments,)
+            If savefig, the times up to which the cumulative 2Fs are computed.
+        twoFs : ndarray of shape (predict_fstat_segments,)
+            If savefig, the cumulative 2F values computed.
         ax : matplotlib.axes._subplots_AxesSubplot, optional
-            If savefig is False
+            If savefig is False, the axes object containing the plot.
 
         """
-        if ax is None:
-            fig, ax = plt.subplots()
-        if injectSources:
+
+        taus, twoFs = self.calculate_twoF_cumulative(**calculate_twoF_cumulative_kwargs)
+        taus_days = taus / 86400.0
+
+        axis_kwargs = {
+            "xlabel": r"Days from $t_{{\rm start}}={:.0f}$".format(
+                calculate_twoF_cumulative_kwargs["tstart"]
+            ),
+            "ylabel": r"$\log_{10}(\mathrm{BSGL})_{\rm cumulative}$"
+            if self.BSGL
+            else r"$\widetilde{2\mathcal{F}}_{\rm cumulative}$",
+            "xlim": (0, taus_days[-1]),
+        }
+        if custom_axis_kwargs is not None:
+            for kwarg in "xlabel", "ylabel":
+                if kwarg in custom_axis_kwargs:
+                    logging.warning(
+                        f"Be careful, overwriting {kwarg} {axis_kwargs[kwarg]}"
+                        " with {custom_axis_kwargs[kwarg]}: Check out the units!"
+                    )
+            axis_kwargs.update(custom_axis_kwargs or {})
+
+        plot_label = plot_label or (
+            f"Cumulative 2F {taus.shape[0]:d} segments"
+            f" ({(taus_days[1] - taus_days[0]):.2g} days per segment)"
+        )
+
+        fig, ax = plt.subplots()
+        ax.grid()
+        ax.set(**axis_kwargs)
+
+        ax.plot(taus_days, twoFs, label=plot_label, color="k")
+
+        if signal_parameters:
```

Comment:
This is what needs to be broken up to make `MCMCSearch.plot_cumulative_max()` happy, I think.

Note I've renamed the argument from `inejction_parameters` to `signal_parameters` because it can just as well be the loudest from a search on real data.